TriangleNet_Engine/Compute/VoronoiRegions.cs

```diff
@@ -0,0 +1,313 @@
+        [Input("tolerance", "Tolerance to be used in the method.", typeof(Length))]
+        [Input("boundarySize", "To handle problems at boundaries, extra points are added outside the bounds of the provided points for the generation of the voronoi and then culled away. This value controls how far off these points should be created. If a zero or negative value is provided, this value will be calculated automatically, based on the size of the boundingbox of the provided points.", typeof(Length))]
+        [Output("regions", "Voronoi regions calculated by the method. The position in the list will correspond to the position in the list of the provided points.")]
+        public static List<Polyline> VoronoiRegions(List<Point> points, Plane plane = null, double boundarySize = -1, double tolerance = Tolerance.Distance)
+        {
+
+            //Preform check all inputs that triangulation can be done
+            if (points == null || points.Count < 3)
+            {
+                Reflection.Compute.RecordError("Insuffient points for triangulation. Please provide at least 3 Points!");
+                return new List<Polyline>();
+            }
+
+            if (points.IsCollinear(tolerance))
+            {
+                return ColinearVoronoiRegions(points, plane, boundarySize, tolerance);
+            }
+
+            //Try fit plane if no is provided
+            if (plane == null)
+            {
+                plane = points.FitPlane(tolerance);
+            }
+
+            if (plane == null)
+            {
+                Engine.Reflection.Compute.RecordError("Could not fit a plane through the Points.");
+                return new List<Polyline>();
+            }
+
+            //Check all points within distance of the plane
+            if (points.Any(x => x.Distance(plane) > tolerance))
+            {
+                BH.Engine.Reflection.Compute.RecordError("Can only handle coplanar points, in the plane!");
+                return new List<Polyline>();
+            }
+
+            //Calculate the local coordinates
+            Vector localX = points[1] - points[0];
+            Vector localY = plane.Normal.CrossProduct(localX);
+            Point min = points.Min();
+            Point refPt = new Point { X = min.X, Y = min.Y };
+            Cartesian localSystem = Create.CartesianCoordinateSystem(min, localX, localY);
+            Cartesian globalSystem = Create.CartesianCoordinateSystem(refPt, Vector.XAxis, Vector.YAxis);
+
+            //Transform to xy-plane
+            List<Point> xyPoints = points.Select(x => x.Orient(localSystem, globalSystem)).ToList();
+
+
+            //Add point at all corners. This is to try to handle weirdness at the boundaries
+            BoundingBox bounds = xyPoints.Bounds();
+
+            double lengthX, lengthY;
```

Comment:
Generally we have avoided this style of variable declaration in the BHoM, and consistency is giving explicit defaults rather than implicit defaults.

So this should probably be:

```
double lengthX = 0;
double lengthY = 0;
```

Reply:
Could probably also do:

```
double lengthX = boundarySize;
double lengthY = boundarySize;
```

Then the `else` part below isn't needed.

Reply:
Generally you might avoided it, but pretty sure I have done this elsewhere. Ofc, both options lead to the same end result, but think the way it is more clearly highligts the intent, with an if statement defining what you should be doing, rather than assigning something and then reassigning if you want something different. I think providing a value here will be an outlier, as most of the time you will be fine with the default (calculated one) and changing the code looks like providing a value will be the default thing you would do.

Not a biggy to fix ofc, but not sure I agree.

Reply:
The first part I am more keen to see different (with separate lines for each variable declaration), the second bit (of assigning to `boundarySize` by default) I am happy to skip based on your explanation above :smile: